src/Memcache.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace TraderInteractive\Memoize;
+
+/**
+ * A memoizer that caches the results in memcache.
+ */
+class Memcache implements Memoize
+{
+    /**
+     * The memcache client
+     *
+     * @var \Memcache
+     */
+    private $client;
+
+    /**
+     * Cache refresh
+     *
+     * @var boolean
+     */
+    private $refresh;
+
+    /**
+     * Sets the memcache client.
+     *
+     * @param \Memcache $client  The memcache client to use
+     * @param boolean         $refresh If true we will always overwrite cache even if it is already set
+     */
+    public function __construct(\Memcache $client, bool $refresh = false)
+    {
+        $this->client = $client;
+        $this->refresh = $refresh;
+    }
+
+    /**
+     * The value is stored in memcache as a json_encoded string,
+     * so make sure that the value you return from $compute is json-encode-able.
+     *
+     * @see Memoize::memoizeCallable
+     *
+     * @param string   $key
+     * @param callable $compute
+     * @param int|null $cacheTime
+     *
+     * @return mixed
+     */
+    public function memoizeCallable(string $key, callable $compute, int $cacheTime = null)
+    {
+        if (!$this->refresh) {
+            try {
+                $cached = $this->client->get($key, $flags, $flags);
+                if ($cached !== false && $cached != null) {
+                    $data = json_decode($cached, true);
+                    return $data['result'];
+                }
+            } catch (\Exception $e) {
+                return call_user_func($compute);
+            }
+        }
+
+        $result = call_user_func($compute);
+
+        // If the result is false/null/empty, then there is no point in storing it in cache.
+        if ($result === false || $result == null || empty($result))
+            return $result;
```

Comment:
Correct to match enforced code standards.
```suggestion
        if ($result === false || $result == null || empty($result)) {
            return $result;
        }
```